sz/Traffic/src/traffic/relogo/UserObserver.groovy

```diff
@@ -54,14 +70,25 @@ class UserObserver extends ReLogoObserver{
 		ask(turtles()){
 			step(dt)
 		}
+		checkDeadlock()
 		resetTimer()
 	}
 	
+	def checkDeadlock() {
+		def isDeadlocked = yieldZones().every { !it.hasRightOfWay() } && userTurtles().every { it.speed == 0 } 
+
+		if (isDeadlocked) {
+			println("Deadlock occurred!")
+			pause() 
+		}
+	}
+	
 	def carsOnTheRoad() {
 		count(userTurtles())
 	}
 	
 	def crashRate() {
+		return UserTurtle.numCrashes
```

Comment:
All lines below are ignored, right?

Reply:
Ah, yes. I found absolute number of crashed cars more useful at some point, but I'll bring the relative rate back, unless you think the current variant makes more sense.